### DISCUSSION TITLE
v2.10.7

### DISCUSSION THREAD

**Discussion by fiftin** [semaphoreui/semaphore#2121] [Announcements]
[![](https://img.shields.io/badge/semaphore_2.10.7_container_configurator-skyblue?style=for-the-badge&logo=docker)](https://semaphoreui.com/install/docker/2_10_7/)

## Features
* Added support for Terraform code, OpenTofu code and Bash scripts.
  
  ![image](https://github.com/semaphoreui/semaphore/assets/914224/e29aea49-dbd4-488b-a7f4-dd71b0af8821)

* Added usage examples.
* Added DEBUG Docker image for remote debugging from Goland.

## Bugfixes
* Fixed infinite "Waiting" status.
* Reduced size of Docker image.
* Fixed issue with authentication on the server with login/password (https://github.com/semaphoreui/semaphore/issues/2098)
* Fixed `web_host` option usage in Docker image.
* Docker Image: Fixed overriding config file options by empty `SEMAPHORE_ACCESS_KEY_ENCRYPTION` and `SEMAPHORE_WEB_ROOT` environment variables.
* Web interface: Removed predefined variables from Environment form.

<hr /><em>This discussion was created from the release <a href='https://github.com/semaphoreui/semaphore/releases/tag/v2.10.7'>v2.10.7</a>.</em>

**Comment by PeterFalken** [comment]
Thank you Denis(@fiftin).
Just finished migrating a deployment on Kubernetes from v2.9.75 to this release without issues.

**Reply by fiftin** [reply]
I'm very glad to hear this!
